src/modules/content-sqlite/content-sqlite.c

```diff
@@ -353,18 +356,28 @@ void checkpoint_get_cb (flux_t *h,
         errno = ENOENT;
         goto error;
     }
+    s = (char *)sqlite3_column_text (ctx->checkpt_get_stmt, 0);
+    if (!(o = json_loads (s, JSON_DECODE_ANY, NULL))) {
+        /* version 0 checkpoint may have been just a blobref string */
```

Comment:
s/may have been just/is/

Could also add a note that version 0 was used prior to flux-core 0.36?

Didn't think of this before, but instead of `json_string(s)` maybe we should just do
```c
o = json_pack ("{s:s s:s s:f}", "version", 0, "value", "rootref", s, "timestamp", 0.);
```
That would simplify the code on the other end.

Reply:
> o = json_pack ("{s:s s:s s:f}", "version", 0, "value", "rootref", s, "timestamp", 0.);

I'm not following.  Perhaps you messed up the json_pack?  I think there's more args than the format can take.

Reply:
Ooops, right - delete "value".  I mean why not construct a consistent "checkpoint object" for version 0 for outside users since it doesn't add any more complexity really.

Reply:
oh wait, i think i know what you meant ... 

```
if (!(o = json_loads (...))) {
      /* if data is a version 0 blobref, respond with version 1 format */
      if (blobref_validate (str))
         o = json_pack ("{s:i s:s s:f}", "version", 0, "rootref", s, "timestamp", 0.);
}
```

Reply:
The title of this PR might be better for the release notes if it ready "enhance KVS checkpoint protocol."